runtime/lua/vim/fcnotify.lua

```diff
@@ -0,0 +1,225 @@
+-- Create a file watcher, each watcher is identified by the name of the file that it
+-- watches.
+
+local uv = vim.loop
+local Watcher = {}
+Watcher.__index = Watcher
+local w = {}
+local WatcherList = {}
+local check_handle = nil
+
+-- Callback for the check handle, checks if there are pending notifications
+-- for any watcher, and handles them as per the value of the `fcnotify`
+-- option.
+local function check_notifications()
+  for _, watcher in pairs(WatcherList) do
+    if watcher.pending_notifs == true then
+      vim.api.nvim_command('checktime '..watcher.bufnr)
+      watcher.pending_notifs = false
+    end
+  end
+end
+
+local function fs_event_start(bufnr)
+  WatcherList[bufnr].handle = uv.new_fs_event()
+  WatcherList[bufnr].handle:start(WatcherList[bufnr].ffname, {}, vim.schedule_wrap(function(...)
+    WatcherList[bufnr]:on_change(...)
+  end))
+end
+
+local function check_handle_start()
+  check_handle = uv.new_check()
+  check_handle:start(vim.schedule_wrap(check_notifications))
+end
+
+local function set_mechanism(option_type, bufnr)
+  if option_type == 'global' then
+    local option = vim.api.nvim_get_option('filechangenotify')
+    if option:find('watcher') then
+      for _, watcher in pairs(WatcherList) do
+        watcher._start_handle = fs_event_start
+      end
+      Watcher.start_notifications = check_handle_start
+    else
+      for _, watcher in pairs(WatcherList) do
+        watcher._start_handle = function(_) end
```

Comment:
You don't need to pass `_` to a function. Lua silently discards mismatched arguments.